File: common/scala_common/redis_manipulate.scala
// ------------------------------------------------------------------    
//	redis_manipulate.scala
//
//					Jan/22/2015
//
// ------------------------------------------------------------------    
import	java.io.PrintWriter
import	java.util.HashMap
import	scala.collection.mutable

import	redis.clients.jedis.Jedis

import org.json4s._
import org.json4s.JsonDSL._
import org.json4s.native.JsonMethods._
// ------------------------------------------------------------------    
object redis_manipulate
{

// ------------------------------------------------------------------    
def redis_to_dict_proc (server:String):(mutable.Map[String,Object]) = {
	var dict_aa = mutable.Map[String,Object] ()

	var jedis = new Jedis(server)

	val set_keys = jedis.keys("*")

	val keys = set_keys.toArray ()

	for (key <- keys)
		{
		val str_key = key.toString ()
		val str_json = jedis.get (str_key)

		if (str_json != null)
			{
//			println (str_json)
			val json = parse (str_json)
			val name = (json \ "name").values
			val population = (json \ "population").values
			val date_mod = (json \ "date_mod").values
			var unit_aa = mutable.Map[String,String] ()
			unit_aa("name") = name.toString
			unit_aa("population") = population.toString
			unit_aa("date_mod") = date_mod.toString
			dict_aa(str_key) = unit_aa
			}
		}
	dict_aa
}

// ------------------------------------------------------------------    
def redis_update_proc (jedis:Jedis,key:String,population_in:Int)
{
	if (jedis.exists(key))
		{
		val str_json = jedis.get(key)
		val	today = text_manipulate.get_current_date_proc ()

		println (str_json)

		val json = parse (str_json)
		val name = (json \ "name").values
		val population = (json \ "population").values
		val date_mod = (json \ "date_mod").values

		var unit_aa = mutable.Map[String,String] ()

		unit_aa("name") = name.toString
		unit_aa("population") = population_in.toString
		unit_aa("date_mod") = today.toString

		val unit_bb = unit_aa.toMap
		val json_str_new = compact(render(unit_bb))

		println (json_str_new);

		jedis.set (key,json_str_new)
	}
}
// ------------------------------------------------------------------    
def dict_to_redis_proc (server:String,dict_aa:mutable.Map[String,Object]){
	var jedis = new Jedis (server)

	for (pair <- dict_aa)
		{
		val key = pair._1
		val unit_aa = pair._2.asInstanceOf [mutable.Map[String,String]]

		val unit_bb = unit_aa.toMap
		val json_str = compact(render(unit_bb))

		println (key)
		println (json_str)

		jedis.set (key,json_str)
		}            
}
// ------------------------------------------------------------------    
}
// ------------------------------------------------------------------    

File: common/java_common/redis_manipulate.java
// ------------------------------------------------------------------    
//	redis_manipulate.java
//
//					Apr/11/2013
//
// ------------------------------------------------------------------    
import	java.io.PrintWriter;
import	java.util.HashMap;
import	java.util.Set;

import	redis.clients.jedis.Jedis;

import	net.arnx.jsonic.JSON;
// ------------------------------------------------------------------    
public class redis_manipulate
{


// ------------------------------------------------------------------    
static void record_parse_proc (String key, String str_json)
{
//	System.out.println (key + "\t" + str_json);
	System.out.print (key + "\t");

	HashMap <String, Object>  map_data = new HashMap <String, Object> ();

	try
	{
	map_data = (HashMap <String, Object>)JSON.decode (str_json);

	String name = map_data.get ("name").toString ();
	String population = map_data.get ("population").toString ();
	String date_mod = map_data.get ("date_mod").toString ();
	System.out.print (name + "\t");
	System.out.print (population + "\t");
	System.out.println (date_mod);
	}
	catch (Exception ee)
	{
	ee.printStackTrace ();
	}

}

// ------------------------------------------------------------------    
static String redis_to_json_proc (Jedis jedis) throws Exception
{
	Set set_keys = jedis.keys("*");

	HashMap <String, HashMap<String,String>> dict_aa
		= new HashMap <String, HashMap<String,String>> ();

	for (Object obj : set_keys)
		{
		String key = obj.toString ();
		String str_json  = jedis.get(key);

		if (str_json != null)
			{
			HashMap <String,String> unit_aa
				= (HashMap <String,String>)JSON.decode (str_json);
			dict_aa.put (key,unit_aa);
			}
		}

	String json_str = json_manipulate.dict_to_json_proc (dict_aa);

	return	json_str;
}

// ------------------------------------------------------------------    
static void redis_update_proc
	(Jedis jedis,String key,int population_in) throws Exception
{
	String str_json = jedis.get(key);

	if (str_json != null)
		{
		redis_manipulate.record_parse_proc (key,str_json);
		HashMap <String,String> unit_aa
			 = (HashMap <String,String>)JSON.decode (str_json);

		System.out.println (unit_aa.get ("name"));        
		String str_population = Integer.toString (population_in);
		String today = text_manipulate.get_current_date_proc ();

		unit_aa.put ("population",str_population);
		unit_aa.put ("date_mod",today);

		String json_new = JSON.encode (unit_aa);

		jedis.set (key,json_new);
		}
}

// ------------------------------------------------------------------    
static String create_new_json_proc (String str_json,int population)
{
	HashMap <String, Object>  unit_aa = (HashMap <String, Object>)JSON.decode (str_json);
        String name = unit_aa.get ("name").toString ();
        String date_mod = text_manipulate.get_current_date_proc ();

	String json_new = json_manipulate.unit_json_gen_proc
		 (name,population,date_mod);

	System.out.println (date_mod);        

	return	json_new;
}

// --------------------------------------------------------------
}
// ------------------------------------------------------------------    
